### PR TITLE
return error if app not found

### LINKED ISSUE #2883: watch command doesn't honor the project or component
/kind bug

## What versions of software are you using?

**Operating System:**

**Output of `odo version`:**
master
## How did you run odo exactly?

no dummy exist for both component or app , but still able to run this command
we should report error

[root@xyimgr77 ftq]# ./odo watch dummy
Waiting for something to change in /root/ftq
^C
[root@xyimgr77 ftq]# ./odo watch --app dummy
Waiting for something to change in /root/ftq

### PR DESCRIPTION
**Pull request by jichenjc**
**What type of PR is this?**

/kind bug

**What does does this PR do / why we need it**:
`odo watch --app adsf` failed to find adsf is invalid

**Which issue(s) this PR fixes**:

Fixes #2883 

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: tests/integration/cmd_watch_test.go [hunk 1]
```diff
@@ -50,6 +50,15 @@ var _ = Describe("odo watch command tests", func() {
 		})
 	})
 
+	Context("when executing watch with wrong app name", func() {
+                It("should fail", func() {
+                        helper.CopyExample(filepath.Join("source", "nodejs"), context)
+                        helper.CmdShouldPass("odo", "component", "create", "nodejs", "--project", project, "--context", context)
+                        output := helper.CmdShouldFail("odo", "watch", "--app", "dummy")
```
Comment (mik-dass) [inline tests/integration/cmd_watch_test.go:57]: you need to use `--context` here for running the command

Comment (jichenjc) [reply tests/integration/cmd_watch_test.go:57]: yes, will update

**Comment by mik-dass** [comment]
The files need to be formatted using gofmt. Also we don't need to pass the app name when using context https://travis-ci.com/github/openshift/odo/jobs/329361159#L458

**Comment by jichenjc** [comment]
can't add integration test due to 
`cannot provide --app, --project or --component flag when --context is provided`

as the test itself is running in some folder while it need --context flag
so I have to remove the intergation test cases

**Comment by mik-dass** [comment]
> can't add integration test due to
cannot provide --app, --project or --component flag when --context is provided

I guess you need to only use the context flag here or am I missing something?

**Comment by jichenjc** [comment]
> 
> 
> > can't add integration test due to
> > cannot provide --app, --project or --component flag when --context is provided
> 
> I guess you need to only use the context flag here or am I missing something?

want to test this: `--app adsf` as I want odo to return error if the app is not found..

**Comment by mik-dass** [comment]
> want to test this: --app adsf as I want odo to return error if the app is not found..

OK let's skip it. @amitkrout WDYT?

**Comment by jichenjc** [comment]
@mik-dass thanks
and I think no plan to support `cd ...; <do some command>` in the test? as it's possible to add test cases if that's supported

**Comment by mik-dass** [comment]
> and I think no plan to support cd ...; <do some command> in the test? as it's possible to add test cases if that's supported

you can try the helper.Chdir(dir) function and then execute the command.

**Comment by jichenjc** [comment]
@mik-dass please check to check whether current change is ok or need further polishment, thanks

**Comment by jichenjc** [comment]
new logic after this PR 

```
[root@bastion frontend]# ../../dist/bin/linux-s390x/odo  watch --app adsf
 ✗  component does not exist. Please use `odo push` to create your component
[root@bastion frontend]# ../../dist/bin/linux-s390x/odo  watch
Waiting for something to change in /root/go/src/github.com/openshift/odo/abc/frontend
^C

```

before change
```
[root@bastion frontend]# odo watch
Waiting for something to change in /root/go/src/github.com/openshift/odo/abc/frontend
^C
[root@bastion frontend]# odo watch --app adsf
Waiting for something to change in /root/go/src/github.com/openshift/odo/abc/frontend
^C

```

**Comment by amitkrout** [comment]
I am verifying the test concern
/hold

**Comment by amitkrout** [comment]
> > want to test this: --app adsf as I want odo to return error if the app is not found..
> 
> OK let's skip it. @amitkrout WDYT?

I am sorry, unfortunately i missed the ping. Anyway we can we can add integration for it. I have created one scenario that verifies this change.

```
Context("when executing odo watch against an app that doesn't exist", func() {
		JustBeforeEach(func() {
			currentWorkingDirectory = helper.Getwd()
			helper.Chdir(context)
		})
		JustAfterEach(func() {
			helper.Chdir(currentWorkingDirectory)
		})
		It("should fail with proper error", func() {
			helper.CopyExample(filepath.Join("source", "nodejs"), context)
			helper.CmdShouldPass("odo", "component", "create", "nodejs", "--project", project)
			output := helper.CmdShouldFail("odo", "watch", "--app", "dummy")
			Expect(output).To(ContainSubstring("component does not exist"))
		})
	})
```
Please use it in the watch integration test file.

**Comment by jichenjc** [comment]
@amitkrout I will add it , thanks a lot

**Comment by jichenjc** [comment]
/test v4.2-integration-e2e
--

**Comment by jichenjc** [comment]
test v4.2-integration-e2e

**Comment by jichenjc** [comment]
Unable to list Service Catalog instances: unable to list ServiceInstances: serviceinstances.servicecatalog.k8s.io is forbidden: User "developer" cannot list resource "serviceinstances" in API group "servicecatalog.k8s.io" in the namespace "dwgxlimbyr"
[odo]

**Comment by jichenjc** [comment]
@amitkrout finally , the test pass... can you help to check it? Thanks

**Pull request merged**